stylix/mk-target.nix

```diff
@@ -130,6 +130,21 @@
       lacks automatic safeguarding and should only be used for complex
       configurations where `configElements` is unsuitable.
 
+    `imports` (List of Modules)
+    : List of Modules to be imported. Note that when using `submodule` imports
+      with NOT have `submodule` applied to them.
+
+    `submodule` (Module -> Module)
+    : A function which transforms a module into a submodule, for example:
+
+      ```nix
+      module: {
+        options.programs.firefox.profiles = lib.mkOption {
+          type = lib.types.attrsOf (lib.types.submodule module);
+        };
+      }
+      ```
```

Comment:
I did not really read the diff, but is there no way around extending the Stylix module system by using existing functionality? It would be sad to increase its complexity unless absolutely necessary.